cmd/lifecycle/builder.go

```diff
@@ -83,12 +83,13 @@ func (ba buildArgs) build(group buildpack.Group, plan platform.BuildPlan) error
 		return cmd.FailErrCode(err, ba.platform.CodeFor(cmd.BuildError), "build")
 	}
 
+	platformAPI := api.MustParse(ba.platform.API())
 	builder := &lifecycle.Builder{
 		AppDir:         ba.appDir,
 		LayersDir:      ba.layersDir,
 		PlatformDir:    ba.platformDir,
-		PlatformAPI:    api.MustParse(ba.platform.API()),
-		Env:            env.NewBuildEnv(os.Environ()),
+		PlatformAPI:    platformAPI,
+		Env:            env.NewBuildEnv(os.Environ(), platformAPI),
```

Comment:
Our use of `platform` is still evolving, but what about passing the entire `platform` to `NewBuildEnv` and on to `isNotIncludedForAPI`? Then we could do something like `if !platform.SupportsAssetPackages { /* return true */ }`

Reply:
I think our long-term dream is for all instances of `platformAPI.Compare(api.MustParse("0.8"))` to end up in the platform package, and the platform interface returns something meaningful instead.